forc-plugins/forc-fmt/src/main.rs

```diff
@@ -63,35 +76,178 @@ fn run() -> Result<()> {
         );
     };
 
-    let dir = match app.path.as_ref() {
-        Some(path) => PathBuf::from(path),
-        None => std::env::current_dir()?,
-    };
-
     let manifest_file = forc_pkg::manifest::ManifestFile::from_dir(&dir)?;
-    for (_, member_path) in manifest_file.member_manifests()? {
-        let member_dir = member_path.dir();
-        let mut formatter = Formatter::from_dir(member_dir)?;
-        format_pkg_at_dir(&app, &dir, &mut formatter)?;
+
+    match manifest_file {
+        ManifestFile::Workspace(ws) => {
+            format_workspace_at_dir(&app, &ws, &dir)?;
+        }
+        ManifestFile::Package(_) => {
+            let mut formatter = Formatter::from_dir(&dir)?;
+            format_pkg_at_dir(&app, &dir, &mut formatter)?;
+        }
     }
 
     Ok(())
 }
 
-fn format_single_file(path: &Path) -> Result<()> {
-    let mut formatter = Formatter::default();
-    if let Ok(file_content) = fs::read_to_string(path) {
-        match Formatter::format(&mut formatter, file_content.into(), None) {
+/// Recursively get a Vec<PathBuf> of subdirectories that contains a Forc.toml.
+fn get_sway_dirs(workspace_dir: PathBuf) -> Vec<PathBuf> {
+    let mut dirs_to_format = vec![];
+    let mut dirs_to_search = vec![workspace_dir];
+
+    while let Some(next_dir) = dirs_to_search.pop() {
+        if let Ok(read_dir) = fs::read_dir(next_dir) {
+            for entry in read_dir.filter_map(|res| res.ok()) {
+                let path = entry.path();
+
+                if path.is_dir() {
+                    dirs_to_search.push(path.clone());
+                    if path.join(constants::MANIFEST_FILE_NAME).exists() {
+                        dirs_to_format.push(path);
+                    }
+                }
+            }
+        }
+    }
+
+    dirs_to_format
+}
+
+/// Format a file, given its path.
+/// Returns:
+/// - Ok(true) if executed successfully and formatted,
+/// - Ok(false) if executed successfully and not formatted,
+/// - Err if it fails to execute at all.
+fn format_file(
+    app: &App,
+    file: PathBuf,
+    manifest_file: Option<PathBuf>,
+    formatter: &mut Formatter,
+) -> Result<bool> {
+    let file = file.canonicalize()?;
+    if let Ok(file_content) = fs::read_to_string(&file) {
+        let mut edited = false;
+        let file_content: Arc<str> = Arc::from(file_content);
+        let build_config = manifest_file.map(|f| {
+            BuildConfig::root_from_file_name_and_manifest_path(
+                file.clone(),
+                f,
+                BuildTarget::default(),
+            )
+        });
+        match Formatter::format(formatter, file_content.clone(), build_config.as_ref()) {
             Ok(formatted_content) => {
-                format_file(path, &formatted_content)?;
+                if app.check {
+                    if *file_content != formatted_content {
+                        info!("\n{:?}\n", file);
```

Comment:
Did you mean to leave this in here? If so, maybe add a prefix to the log message

```suggestion
                        info!("File was edited by formatter: \n{:?}\n", file);
```